File: Frontend/src/types/conversation.ts
export interface ConversationResponse {
  conversation_id: string;
  name: string | null;
  is_group: boolean;
  created_at: string;
}

export interface ConversationMemberInfo {
  member_id: string;
  joined_at: string;
}

export interface ConversationMembersResponse {
  conversation_id: string;
  members: ConversationMemberInfo[];
}

export interface CreateConversation {
  name: string | null;
  is_group: boolean;
  member_ids: string[];
}

File: api/src/dtos/conversation.rs
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};
use uuid::Uuid;

#[derive(Serialize)]
pub struct ConversationResponse {
    pub conversation_id: Uuid,
    pub name: Option<String>,
    pub is_group: bool,
    pub created_at: DateTime<Utc>,
}

#[derive(Serialize)]
pub struct ConversationMemberInfo {
    pub member_id: Uuid,
    pub joined_at: DateTime<Utc>,
}

#[derive(Serialize)]
pub struct ConversationMembersResponse {
    pub conversation_id: Uuid,
    pub members: Vec<ConversationMemberInfo>,
}

#[derive(Deserialize)]
pub struct CreateConversation {
    pub name: Option<String>,
    pub is_group: bool,
    pub member_ids: Vec<Uuid>,
}
